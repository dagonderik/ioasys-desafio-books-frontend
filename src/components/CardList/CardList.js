import React from 'react';
import Cards from '../Cards/Cards.js';
import PropTypes from 'prop-types'

/**
 * Component used to create the List of cards displaying the books
 * 
 */

const CardList = ({bookList, onRouteChange}) => {
    return (
        <div>
            {
                bookList.map((user,i) => {
                    return (
                        <Cards
                            key={bookList[i].id}
                            onRouteChange={onRouteChange}    
                            id={i}
                            authors={bookList[i].authors}
                            title={bookList[i].title}
                            pageCount={bookList[i].pageCount}
                            publisher={bookList[i].publisher}
                            published={bookList[i].published}
                            imageUrl={bookList[i].imageUrl}
                        />
                    );
                })
            }
        </div>
    )
}

CardList.propTypes = {
    /**
     * bookList holds the list of books got from the API
     */
    bookList: PropTypes.array,
    /**
     * onRouteChange is a funtion that receives a value to determine which screen to show
     */
    onRouteChange: PropTypes.func
}

export default CardList;