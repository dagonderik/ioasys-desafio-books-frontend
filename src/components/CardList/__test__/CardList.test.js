import React from "react";
import ReactDOM from 'react-dom';
import renderer from "react-test-renderer";
import CardList from "./../CardList";

const bookTest=[
    {
        id:1,
        authors:["authors"],
        title:"title",
        pageCount:1,
        publisher:"publisher",
        published:"published",
        imageUrl:"imageURL"
    },
    {
        id:2,
        authors:["authors"],
        title:"title",
        pageCount:1,
        publisher:"publisher",
        published:"published",
        imageUrl:"imageURL"
    },
    {
        id:3,
        authors:["authors"],
        title:"title",
        pageCount:1,
        publisher:"publisher",
        published:"published",
        imageUrl:"imageURL"
    }

]

it("renders without crashing", () => {  
    const div = document.createElement("div");
    ReactDOM.render(<CardList 
    bookList={bookTest}
    onRouteChange={()=>{}}></CardList>,div)
})

it("matches snapshot", () =>{
    const tree = renderer.create(<CardList 
    bookList={bookTest}
    onRouteChange={()=>{}}></CardList>).toJSON();
    expect(tree).toMatchSnapshot();
})