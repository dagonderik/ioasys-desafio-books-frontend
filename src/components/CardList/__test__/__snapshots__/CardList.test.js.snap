// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    className="card tc white dib br3 pa3 ma2 pointer grow shadow-5 card"
    data-testid="Cards"
    onClick={[Function]}
    style={
      Object {
        "height": "auto",
        "margin": "10px",
        "maxWidth": "500px",
        "minWidth": "400px",
        "textAlign": "left",
      }
    }
  >
    <img
      alt=""
      src="imageURL"
      style={
        Object {
          "align": "center",
          "display": "inline-block",
          "height": "150px",
          "verticalAlign": "middle",
        }
      }
    />
    <div
      style={
        Object {
          "display": "inline-block",
          "marginLeft": "10px",
          "verticalAlign": "middle",
        }
      }
    >
      <h2
        className="black f4"
      >
        title
      </h2>
      <p
        className="red f7"
      >
        authors
        <br />
      </p>
      <p
        className="black-50 f7"
      >
         
        1
         
         páginas
         
        <br />
        publisher
        <br />
        published
      </p>
    </div>
  </div>
  <div
    className="card tc white dib br3 pa3 ma2 pointer grow shadow-5 card"
    data-testid="Cards"
    onClick={[Function]}
    style={
      Object {
        "height": "auto",
        "margin": "10px",
        "maxWidth": "500px",
        "minWidth": "400px",
        "textAlign": "left",
      }
    }
  >
    <img
      alt=""
      src="imageURL"
      style={
        Object {
          "align": "center",
          "display": "inline-block",
          "height": "150px",
          "verticalAlign": "middle",
        }
      }
    />
    <div
      style={
        Object {
          "display": "inline-block",
          "marginLeft": "10px",
          "verticalAlign": "middle",
        }
      }
    >
      <h2
        className="black f4"
      >
        title
      </h2>
      <p
        className="red f7"
      >
        authors
        <br />
      </p>
      <p
        className="black-50 f7"
      >
         
        1
         
         páginas
         
        <br />
        publisher
        <br />
        published
      </p>
    </div>
  </div>
  <div
    className="card tc white dib br3 pa3 ma2 pointer grow shadow-5 card"
    data-testid="Cards"
    onClick={[Function]}
    style={
      Object {
        "height": "auto",
        "margin": "10px",
        "maxWidth": "500px",
        "minWidth": "400px",
        "textAlign": "left",
      }
    }
  >
    <img
      alt=""
      src="imageURL"
      style={
        Object {
          "align": "center",
          "display": "inline-block",
          "height": "150px",
          "verticalAlign": "middle",
        }
      }
    />
    <div
      style={
        Object {
          "display": "inline-block",
          "marginLeft": "10px",
          "verticalAlign": "middle",
        }
      }
    >
      <h2
        className="black f4"
      >
        title
      </h2>
      <p
        className="red f7"
      >
        authors
        <br />
      </p>
      <p
        className="black-50 f7"
      >
         
        1
         
         páginas
         
        <br />
        publisher
        <br />
        published
      </p>
    </div>
  </div>
</div>
`;
